### PR TITLE
change perl-XML-SAX tarball generation script so it can be used by autopugrade tool

### PR DESCRIPTION
**Pull request by nicogbg**
###### Merge Checklist  

**All** boxes should be checked before merging the PR *(just tick any boxes which don't apply to this PR)*
- [x] The toolchain has been rebuilt successfully (or no changes were made to it)
- [x] The toolchain/worker package manifests are up-to-date
- [x] Any updated packages successfully build (or no packages were changed)
- [x] Packages depending on static components modified in this PR (Golang, `*-static` subpackages, etc.) have had their `Release` tag incremented.
- [x] Package tests (%check section) have been verified with RUN_CHECK=y for existing SPEC files, or added to new SPEC files
- [x] All package sources are available
- [x] cgmanifest files are up-to-date and sorted (`./cgmanifest.json`, `./toolkit/tools/cgmanifest.json`, `./toolkit/scripts/toolchain/cgmanifest.json`, `.github/workflows/cgmanifest.json`)
- [x] LICENSE-MAP files are up-to-date (`./SPECS/LICENSES-AND-NOTICES/data/licenses.json`, `./SPECS/LICENSES-AND-NOTICES/LICENSES-MAP.md`, `./SPECS/LICENSES-AND-NOTICES/LICENSE-EXCEPTIONS.PHOTON`)
- [x] All source files have up-to-date hashes in the `*.signatures.json` files
- [x] `sudo make go-tidy-all` and `sudo make go-test-coverage` pass
- [x] Documentation has been updated to match any changes to the build system
- [x] Ready to merge

---

###### Summary 

change perl-XML-SAX tarball generation script so it can be used by autopugrade tool

###### Change Log  

- Change
- Change
- Change

###### Does this affect the toolchain?  

**NO**

###### Associated issues  

- #xxxx

###### Links to CVEs  
- https://nvd.nist.gov/vuln/detail/CVE-YYYY-XXXX

###### Test Methodology

- Pipeline build id: xxxx

### CODE REVIEW

File: SPECS/perl-XML-SAX/generate_source_tarball.sh [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        *) # preserve positional arguments
+        PARAMS="$PARAMS $1"
+        shift
+        ;;
+  esac
+done
+
+echo "--srcTarball   -> $SRC_TARBALL"
+echo "--outFolder    -> $OUT_FOLDER"
+echo "--pkgVersion   -> $PKG_VERSION"
```
Comment (oliviacrain) [inline SPECS/perl-XML-SAX/generate_source_tarball.sh:59]: It looks like this script still uses $VERSION throughout, which is undefined.

Comment (nicogbg) [reply SPECS/perl-XML-SAX/generate_source_tarball.sh:59]: arggg, Thanks !!

File: SPECS/perl-XML-SAX/generate_source_tarball.sh [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+
+if [ -z "$PKG_VERSION" ]; then
+    echo "--pkgVersion parameter cannot be empty"
+    exit 1
+fi
+
+if [ -z "$SRC_TARBALL" ]; then
+    echo "--srcTarball parameter cannot be empty"
+    exit 1
+fi
```
[the remark this answers is not in the document]
Comment (nicogbg) [reply SPECS/perl-XML-SAX/generate_source_tarball.sh]: done

File: SPECS/perl-XML-SAX/perl-XML-SAX.signatures.json [hunk 3]
```diff
@@ -1,6 +1,5 @@
 {
  "Signatures": {
-  "XML-SAX-1.02-nopatents.tar.gz": "044404d4fdb52485bf143e2661eb4b59de955d4805231befad1bb41bfb715a6f",
-  "generate-tarball.sh": "d360e039412451e0596d4cf0ae9a072d15f94ba0beb4c0740c3ca5340de6f5b2"
+  "XML-SAX-1.02.tar.gz": "044404d4fdb52485bf143e2661eb4b59de955d4805231befad1bb41bfb715a6f"
```
Comment (mandeepsplaha) [inline SPECS/perl-XML-SAX/perl-XML-SAX.signatures.json:3]: Do we not need an entry for the `generate_source_tarball.sh` script?

**Review by oliviacrain — APPROVED** [review]

**Review by mandeepsplaha — APPROVED** [review]

**Pull request merged**